infrastructure/stage/step-functions/index.ts

```diff
@@ -243,6 +243,17 @@ function wireUpStateMachinePermissions(scope: Construct, props: SfnPropsWithStat
     );
   }
 
+  if (stepFunctionsRequirementsMap[props.stateMachineName].needsJobsConfigReadPermissions) {
```

Comment:
I think `stepFunctionsRequirementsMap[props.stateMachineName]` can be reduced to `sfnRequirements`?